### ISSUE TITLE
npm install fails 

### ISSUE THREAD

**Issue by nichoth** [dominictarr/level-sublevel#104]
```
$ node --version
v5.0.0

$ sw_vers
ProductName:	Mac OS X
ProductVersion:	10.10.5
BuildVersion:	14F1509

$ npm --version
3.3.6
```

Errors

```
CXX(target) Release/obj.target/leveldb/deps/leveldb/leveldb-1.14.0/util/env_posix.o
  LIBTOOL-STATIC Release/leveldb.a
  CXX(target) Release/obj.target/snappy/deps/snappy/snappy-1.1.1/snappy-sinksource.o
  CXX(target) Release/obj.target/snappy/deps/snappy/snappy-1.1.1/snappy-stubs-internal.o
  CXX(target) Release/obj.target/snappy/deps/snappy/snappy-1.1.1/snappy.o
  LIBTOOL-STATIC Release/snappy.a
  CXX(target) Release/obj.target/leveldown/src/batch.o
In file included from ../src/batch.cc:5:
../../../../nan/nan.h:261:25: error: redefinition of '_NanEnsureLocal'
NAN_INLINE v8::Local<T> _NanEnsureLocal(v8::Local<T> val) {
                        ^
../../../../nan/nan.h:256:25: note: previous definition is here
NAN_INLINE v8::Local<T> _NanEnsureLocal(v8::Handle<T> val) {
                        ^
../../../../nan/nan.h:661:13: error: no member named 'smalloc' in namespace
      'node'
    , node::smalloc::FreeCallback callback
      ~~~~~~^
../../../../nan/nan.h:672:12: error: no matching function for call to 'New'
    return node::Buffer::New(v8::Isolate::GetCurrent(), data, size);
           ^~~~~~~~~~~~~~~~~
/Users/nick/.node-gyp/5.0.0/include/node/node_buffer.h:31:40: note: candidate
      function not viable: no known conversion from 'uint32_t'
      (aka 'unsigned int') to 'enum encoding' for 3rd argument
NODE_EXTERN v8::MaybeLocal<v8::Object> New(v8::Isolate* isolate,
                                       ^
/Users/nick/.node-gyp/5.0.0/include/node/node_buffer.h:43:40: note: candidate
      function not viable: 2nd argument ('const char *') would lose const
      qualifier
NODE_EXTERN v8::MaybeLocal<v8::Object> New(v8::Isolate* isolate,
                                       ^
/Users/nick/.node-gyp/5.0.0/include/node/node_buffer.h:28:40: note: candidate
      function not viable: requires 2 arguments, but 3 were provided
NODE_EXTERN v8::MaybeLocal<v8::Object> New(v8::Isolate* isolate, size_t length);
                                       ^
/Users/nick/.node-gyp/5.0.0/include/node/node_buffer.h:36:40: note: candidate
      function not viable: requires 5 arguments, but 3 were provided
NODE_EXTERN v8::MaybeLocal<v8::Object> New(v8::Isolate* isolate,
                                       ^
In file included from ../src/batch.cc:5:
../../../../nan/nan.h:676:12: error: no viable conversion from
      'v8::MaybeLocal<v8::Object>' to 'v8::Local<v8::Object>'
    return node::Buffer::New(v8::Isolate::GetCurrent(), size);
           ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/Users/nick/.node-gyp/5.0.0/include/node/v8.h:210:7: note: candidate constructor
      (the implicit copy constructor) not viable: no known conversion from
      'v8::MaybeLocal<v8::Object>' to 'const v8::Local<v8::Object> &' for 1st
      argument
class Local {
      ^
/Users/nick/.node-gyp/5.0.0/include/node/v8.h:210:7: note: candidate constructor
      (the implicit move constructor) not viable: no known conversion from
      'v8::MaybeLocal<v8::Object>' to 'v8::Local<v8::Object> &&' for 1st
      argument
class Local {
      ^
/Users/nick/.node-gyp/5.0.0/include/node/v8.h:214:13: note: candidate template
      ignored: could not match 'Local' against 'MaybeLocal'
  V8_INLINE Local(Local<S> that)
            ^
/Users/nick/.node-gyp/5.0.0/include/node/v8.h:326:13: note: candidate template
      ignored: could not match 'S *' against 'v8::MaybeLocal<v8::Object>'
  V8_INLINE Local(S* that)
            ^
In file included from ../src/batch.cc:5:
../../../../nan/nan.h:683:26: error: no member named 'Use' in namespace
      'node::Buffer'
    return node::Buffer::Use(v8::Isolate::GetCurrent(), data, size);
           ~~~~~~~~~~~~~~^
In file included from ../src/batch.cc:1:
In file included from /Users/nick/.node-gyp/5.0.0/include/node/node.h:42:
/Users/nick/.node-gyp/5.0.0/include/node/v8.h:221:5: error: assigning to
      'v8::Primitive *volatile' from incompatible type 'v8::Value *'
    TYPE_CHECK(T, S);
    ^~~~~~~~~~~~~~~~
/Users/nick/.node-gyp/5.0.0/include/node/v8.h:180:37: note: expanded from macro
      'TYPE_CHECK'
    *(static_cast<T* volatile*>(0)) = static_cast<S*>(0);      \
                                    ^ ~~~~~~~~~~~~~~~~~~
../../../../nan/nan.h:414:12: note: in instantiation of function template
      specialization 'v8::Local<v8::Primitive>::Local<v8::Value>' requested here
    return NanEscapeScope(NanNew(v8::Undefined(v8::Isolate::GetCurrent())));
           ^
../../../../nan/nan.h:398:30: note: expanded from macro 'NanEscapeScope'
# define NanEscapeScope(val) scope.Escape(_NanEnsureLocal(val))
                             ^
In file included from ../src/batch.cc:1:
In file included from /Users/nick/.node-gyp/5.0.0/include/node/node.h:42:
/Users/nick/.node-gyp/5.0.0/include/node/v8.h:221:5: error: assigning to
      'v8::Boolean *volatile' from incompatible type 'v8::Value *'
    TYPE_CHECK(T, S);
    ^~~~~~~~~~~~~~~~
/Users/nick/.node-gyp/5.0.0/include/node/v8.h:180:37: note: expanded from macro
      'TYPE_CHECK'
    *(static_cast<T* volatile*>(0)) = static_cast<S*>(0);      \
                                    ^ ~~~~~~~~~~~~~~~~~~
../../../../nan/nan.h:424:12: note: in instantiation of function template
      specialization 'v8::Local<v8::Boolean>::Local<v8::Value>' requested here
    return NanEscapeScope(NanNew(v8::True(v8::Isolate::GetCurrent())));
           ^
../../../../nan/nan.h:398:30: note: expanded from macro 'NanEscapeScope'
# define NanEscapeScope(val) scope.Escape(_NanEnsureLocal(val))
                             ^
In file included from ../src/batch.cc:1:
In file included from /Users/nick/.node-gyp/5.0.0/include/node/node.h:42:
/Users/nick/.node-gyp/5.0.0/include/node/v8.h:221:5: error: assigning to
      'v8::Function *volatile' from incompatible type 'v8::Value *'
    TYPE_CHECK(T, S);
    ^~~~~~~~~~~~~~~~
/Users/nick/.node-gyp/5.0.0/include/node/v8.h:180:37: note: expanded from macro
      'TYPE_CHECK'
    *(static_cast<T* volatile*>(0)) = static_cast<S*>(0);      \
                                    ^ ~~~~~~~~~~~~~~~~~~
../../../../nan/nan.h:1514:12: note: in instantiation of function template
      specialization 'v8::Local<v8::Function>::Local<v8::Value>' requested here
    return NanEscapeScope(NanNew(handle)->Get(kCallbackIndex)
           ^
../../../../nan/nan.h:398:30: note: expanded from macro 'NanEscapeScope'
# define NanEscapeScope(val) scope.Escape(_NanEnsureLocal(val))
                             ^
In file included from ../src/batch.cc:1:
In file included from /Users/nick/.node-gyp/5.0.0/include/node/node.h:42:
/Users/nick/.node-gyp/5.0.0/include/node/v8.h:221:5: error: assigning to
      'v8::Object *volatile' from incompatible type 'v8::Value *'
    TYPE_CHECK(T, S);
    ^~~~~~~~~~~~~~~~
/Users/nick/.node-gyp/5.0.0/include/node/v8.h:180:37: note: expanded from macro
      'TYPE_CHECK'
    *(static_cast<T* volatile*>(0)) = static_cast<S*>(0);      \
                                    ^ ~~~~~~~~~~~~~~~~~~
../../../../nan/nan.h:1632:12: note: in instantiation of function template
      specialization 'v8::Local<v8::Object>::Local<v8::Value>' requested here
    return NanEscapeScope(handle->Get(NanNew(key)).As<v8::Object>());
           ^
../../../../nan/nan.h:398:30: note: expanded from macro 'NanEscapeScope'
# define NanEscapeScope(val) scope.Escape(_NanEnsureLocal(val))
                             ^
9 errors generated.
make: *** [Release/obj.target/leveldown/src/batch.o] Error 1
gyp ERR! build error 
gyp ERR! stack Error: `make` failed with exit code: 2
gyp ERR! stack     at ChildProcess.onExit (/usr/local/lib/node_modules/npm/node_modules/node-gyp/lib/build.js:270:23)
gyp ERR! stack     at emitTwo (events.js:87:13)
gyp ERR! stack     at ChildProcess.emit (events.js:172:7)
gyp ERR! stack     at Process.ChildProcess._handle.onexit (internal/child_process.js:200:12)
gyp ERR! System Darwin 14.5.0
gyp ERR! command "/usr/local/bin/node" "/usr/local/lib/node_modules/npm/node_modules/node-gyp/bin/node-gyp.js" "rebuild"
gyp ERR! cwd /Users/nick/code/clones/level-sublevel/node_modules/level/node_modules/leveldown
gyp ERR! node -v v5.0.0
gyp ERR! node-gyp -v v3.0.3
gyp ERR! not ok 
npm info lifecycle leveldown@0.10.5~install: Failed to exec install script
level-sublevel@6.5.2 /Users/nick/code/clones/level-sublevel
├─┬ bytewise@0.7.1 
│ └─┬ bops@0.1.1 
│   ├── base64-js@0.0.2 
│   └── to-utf8@0.0.1 
├── level-packager@0.18.0  extraneous
├─┬ level-test@1.7.0 
│ ├─┬ level-js@2.1.6 
│ │ ├─┬ abstract-leveldown@0.12.4 
│ │ │ └── xtend@3.0.0 
│ │ ├── idb-wrapper@1.4.1 
│ │ ├── isbuffer@0.0.0 
│ │ ├── ltgt@1.2.0 
│ │ ├─┬ tape@2.10.3 
│ │ │ └── object-inspect@0.3.1 
│ │ ├── typedarray-to-buffer@1.0.4 
│ │ └─┬ xtend@2.1.2 
│ │   └── object-keys@0.4.0 
│ ├── leveldown@0.10.5 
│ ├─┬ memdown@0.10.2 
│ │ └── ltgt@1.0.2 
│ ├─┬ mkdirp@0.5.1 
│ │ └── minimist@0.0.8 
│ ├── osenv@0.1.0 
│ └── rimraf@2.2.8 
├─┬ leveldown@1.0.7 
│ ├── abstract-leveldown@2.2.2 
│ ├── bindings@1.2.1 
│ ├── fast-future@1.0.1 
│ └── nan@1.8.4 
├─┬ levelup@0.19.0 
│ ├── bl@0.8.2 
│ ├── deferred-leveldown@0.2.0 
│ ├── errno@0.1.4 
│ ├── prr@0.0.0 
│ ├─┬ readable-stream@1.0.33 
│ │ ├── core-util-is@1.0.2 
│ │ ├── isarray@0.0.1 
│ │ └── string_decoder@0.10.31 
│ ├── semver@2.3.2 
│ └── xtend@3.0.0 
├── ltgt@2.1.2 
├── monotonic-timestamp@0.0.8 
├─┬ pull-level@1.1.12 
│ ├─┬ level-post@1.0.4 
│ │ ├── buffer-equal@0.0.1 
│ │ └── string-range@1.2.2 
│ ├── pull-cat@1.1.8 
│ ├─┬ pull-pushable@1.1.4 
│ │ └── pull-stream@2.18.3 
│ ├─┬ pull-window@2.1.3 
│ │ └── looper@2.0.0 
│ └── stream-to-pull-stream@1.3.1 
├─┬ pull-stream@2.21.0 
│ └── pull-core@1.0.0 
├─┬ rimraf@2.1.4 
│ └── graceful-fs@1.2.3 
├── shasum@0.0.2 
├── stream-to-pull-stream@1.2.3 
├─┬ tape@2.14.0 
│ ├── deep-equal@0.2.2 
│ ├── defined@0.0.0 
│ ├─┬ glob@3.2.11 
│ │ └─┬ minimatch@0.3.0 
│ │   ├── lru-cache@2.7.3 
│ │   └── sigmund@1.0.1 
│ ├── has@0.0.1 
│ ├── inherits@2.0.1 
│ ├── object-inspect@1.1.0 
│ └── resumer@0.0.0 
├── through@2.3.8 
├── typewiselite@1.0.0 
└── xtend@4.0.1 

npm ERR! Darwin 14.5.0
npm ERR! argv "/usr/local/bin/node" "/usr/local/bin/npm" "install"
npm ERR! node v5.0.0
npm ERR! npm  v3.3.6
npm ERR! code ELIFECYCLE

npm ERR! leveldown@0.10.5 install: `node-gyp rebuild`
npm ERR! Exit status 1
npm ERR! 
npm ERR! Failed at the leveldown@0.10.5 install script 'node-gyp rebuild'.
npm ERR! This is most likely a problem with the leveldown package,
npm ERR! not with npm itself.
npm ERR! Tell the author that this fails on your system:
npm ERR!     node-gyp rebuild
npm ERR! You can get their info via:
npm ERR!     npm owner ls leveldown
npm ERR! There is likely additional logging output above.

npm ERR! Please include the following file with any support request:
npm ERR!     /Users/nick/code/clones/level-sublevel/npm-debug.log
```

**Comment by swashcap** [comment]
I’m seeing this on OS X 10.11.2 for both these environments:

* Node 4.2.3, NPM 2.14.7
* Node 5.3.0, NPM 3.3.12

Bumping [leveldown](https://www.npmjs.com/package/leveldown) to 1.4.3 makes it successfully build.

**Comment by ralphtheninja** [comment]
Working on this now.

**Comment by ralphtheninja** [comment]
Hmm installing `level-sublevel` from npm works for me, so this is only a problem when installing `devDependencies`

**Comment by dominictarr** [comment]
fixed in 6.5.4
